.github/workflows/dependabot-automerge.yml

```diff
@@ -0,0 +1,23 @@
+name: Dependabot auto-merge
+on: pull_request
+
+permissions:
+  pull-requests: write
+  contents: write
+
+jobs:
+  dependabot:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    steps:
+      - name: Dependabot metadata
+        id: metadata
+        uses: dependabot/fetch-metadata@v1.1.1
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
```

Comment:
what privileges does this token have?

Reply:
This is what's suggested by GitHub - https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request

Reply:
@petemoore I've made all checks required now. Refresh this PR to see the required badge next to all the tasks that run.

Reply:
I _believe_ this should respect our setting ` Require branches to be up to date before merging` we have enabled.